radixdlt-core/radixdlt/src/main/java/com/radixdlt/api/core/model/CoreModelMapper.java

```diff
@@ -0,0 +1,1207 @@
+import com.radixdlt.utils.Pair;
+import com.radixdlt.utils.PrivateKeys;
+import com.radixdlt.utils.UInt256;
+import com.radixdlt.utils.UInt384;
+
+import java.math.BigInteger;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.function.Function;
+
+public final class CoreModelMapper {
+	private static final String TOKEN_TYPE = "Token";
+	private static final String STAKE_UNIT_TYPE = "StakeUnit";
+	private static final String SYSTEM_ADDRESS = "system";
+	private static final String PREPARED_STAKES_ADDRESS = "prepared_stakes";
+	private static final String PREPARED_UNSTAKES_ADDRESS = "prepared_unstakes";
+	private static final String EXITING_UNSTAKES_ADDRESS = "exiting_unstakes";
+	private static final ECPublicKey MOCK_PUBLIC_KEY = PrivateKeys.ofNumeric(1).getPublicKey();
+	private final Addressing addressing;
+	private final Network network;
+	private final Forks forks;
+
+	@Inject
+	CoreModelMapper(
+		@NetworkId int networkId,
+		Addressing addressing,
+		Forks forks
+	) {
+		this.network = Network.ofId(networkId).orElseThrow();
+		this.addressing = addressing;
+		this.forks = forks;
+	}
+
+	public void verifyNetwork(NetworkIdentifier networkIdentifier) throws CoreApiException {
+		if (!networkIdentifier.getNetwork().equals(this.network.name().toLowerCase())) {
+			throw CoreApiException.notSupported(
+				new NetworkNotSupportedError()
+					.addSupportedNetworksItem(new NetworkIdentifier().network(this.network.name().toLowerCase()))
+					.type(NetworkNotSupportedError.class.getSimpleName())
+			);
+		}
+	}
+
+	public Pair<ECPublicKey, ECDSASignature> keyAndSignature(Signature signature) throws CoreApiException {
+		var bytes = bytes(signature.getBytes());
+		ECDSASignature sig;
+		try {
+			sig = ECDSASignature.decodeFromDER(bytes);
+		} catch (IllegalArgumentException e) {
+			throw CoreApiException.badRequest(
+				new InvalidSignatureError()
+					.invalidSignature(signature.getBytes())
+					.type(InvalidSignatureError.class.getSimpleName())
+			);
+		}
+		var publicKey = ecPublicKey(signature.getPublicKey());
+		return Pair.of(publicKey, sig);
+	}
+
+	public ECPublicKey ecPublicKey(PublicKey publicKey) throws CoreApiException {
+		var bytes = bytes(publicKey.getHex());
+		try {
+			return ECPublicKey.fromBytes(bytes);
+		} catch (PublicKeyException e) {
+			throw CoreApiException.badRequest(new InvalidPublicKeyError()
+				.invalidPublicKey(publicKey)
+				.type(InvalidPublicKeyError.class.getSimpleName())
+			);
+		}
+	}
+
+	public byte[] bytes(String hex) throws CoreApiException {
+		try {
+			return Bytes.fromHexString(hex);
+		} catch (IllegalArgumentException e) {
+			throw CoreApiException.badRequest(new InvalidHexError()
+				.invalidHex(hex)
+				.type(InvalidHexError.class.getSimpleName())
+			);
+		}
+	}
+
+	public CoreError builderErrorDetails(TxBuilderException e) {
+		if (e instanceof MinimumStakeException minimumStakeException) {
```

Comment:
is there a way to improve these long if/else? If there isn't an easy way of doing it now I would add a task to do that as soon as possible.

Reply:
Yes, I think there is a better of handling this but would require lower level refactors not for this PR. Will add a task for this.